crypto/fipsmodule/ec/ec_key.c

```diff
@@ -550,9 +550,8 @@ int EC_KEY_generate_key_fips(EC_KEY *eckey) {
 
 #if defined(AWSLC_FIPS)
   AWS_LC_FIPS_failure("EC keygen checks failed");
-#else
-  return 0;
 #endif
+  return 0;
```

Comment:
Does this affect any unreachable-code analysis? As the `return 0;` could be seen as unreachable due to `AWS_LC_FIPS_failure` trigger an abort?

Reply:
In theory yes, but nothing seems to be catching the issue. Looks like GCC removed support for the check a while ago https://gcc.gnu.org/bugzilla/show_bug.cgi?id=82100#c1, Clang has an option but we don't have it turned on https://clang.llvm.org/docs/DiagnosticsReference.html#wunreachable-code. 

I enabled that option for clang and it correctly flags the `exit(1)` is unreachable after the abort. But it doesn't catch this new unreachable code. Reading the bug reports it looks like detecting unreachable code is a hard problem and very dependent on compiler versions and optimizations.

Debug build:
```
bcm.c:398:5: error: code will never be executed [-Werror,-Wunreachable-code]
  398 |     exit(1);
      |     ^~~~
```

Release build didn't find anything else with Clang.